docs/guides/app-dev/mempool-lanes.md

```diff
@@ -0,0 +1,163 @@
+setting up and using Lanes within your application. 
+
+## How to set up lanes in an application
+
+Application developers that want to make use of Lanes must pre-define a list of lane names and their
+priorities and then populate specific fields in `Info` and `CheckTx` responses. In contrast, if
+Lanes are not to be used by the application, no modifications to the code are necessary.
+
+We will explain with an example taken from our implementation of Lanes in the `kvstore` application
+(in `abci/example/kvstore/kvstore.go`).
+
+### Define lanes and their priorities
+
+First, the application should keep a list of lane IDs (of type `string`) and their priorities (of
+type `uint32`). In this example we store it as a map in the `Application` struct. And we define as a
+constant the lane ID used as default when assigning lanes to transactions.
+
+```go
+const defaultLane = "C"
+
+type Application struct {
+  ...
+  lanePriorities map[string]uint32
+}
+
+func NewApplication(...) *Application {
+  ...
+  return &Application{
+    ...
+    lanePriorities: map[string]uint32{
+      "A": 100,
+      "B": 50,
+      defaultLane: 10,
+      "D": 1,
+    },
+  }
+}
+```
+
+### Handling Info requests
+
+When a CometBFT node starts, it performs a handshake with the application by sending an `Info`
+request. This process allows CometBFT to retrieve essential data from the application to initialize
+and configure itself.
+
+Upon receiving an `Info` request, the application must reply with the lane IDs and
+their priorities in the `LanePriorities` field, and the default lane in the `DefaultLane` field. The
+default lane ID must be a key in the map `LanePriorities`.
+
+```go
+func (app *Application) Info(ctx context.Context, req *types.InfoRequest) (*types.InfoResponse, error) {
+  ...
+  return &types.InfoResponse{
+    ...
+    LanePriorities:   app.lanePriorities,
+    DefaultLane:      defaultLane,
+  }, nil
+}
+```
+
+If the application does not populate these fields, CometBFT will use a single, default lane.
+
+### Handling CheckTx requests
+
+Upon receiving a `CheckTx` request for validating a transaction, the application must reply with the
+lane that it assigns to the transaction in the `LaneId` field. The mempool will only use the lane if
+(1) the transaction is valid and (2) the transaction is being validated for the first time at the
+local node, that is, when `req.Type` equals `types.CHECK_TX_TYPE_CHECK` (not when rechecking).
+Otherwise, the mempool will ignore the `LaneId` field.
+
+```go
+func (app *Application) CheckTx(ctx context.Context, req *types.CheckTxRequest) (*types.CheckTxResponse, error) {
+  ...
+  laneID := assignLane(req.Tx)
+  return &types.CheckTxResponse{Code: CodeTypeOK, GasWanted: 1, LaneId: laneID}, nil
+}
+```
+In this example, `assignLane` is a deterministic function that, given the content of a transaction,
+returns a valid lane ID. The lane ID must be one of the keys in the `app.lanePriorities` map, and it
```

Comment:
Actually, should be one of the keys in the `LanePriorities` field returned by `Info`.